filebeat/fileset/pipelines.go

```diff
@@ -129,6 +135,37 @@ func loadPipeline(esClient PipelineLoader, pipelineID string, content map[string
 	return nil
 }
 
+// setEcsProcessors sets required ECS options in processors when filebeat version is 7.0.X
+// and ES is 6.7.X to ease migration to ECS
+func setEcsProcessors(esVersion common.Version, pipelineID string, content map[string]interface{}) error {
+	ecsVersion := common.MustNewVersion("7.0.0")
+	if !esVersion.LessThan(ecsVersion) {
+		return nil
+	}
+
+	processors, ok := content["processors"].([]interface{})
+	if !ok {
+		logp.Debug("modules", "Pipeline '%s' without processors: %+v", pipelineID, content)
```

Comment:
This log message isn't always accurate. Let's split the the checks for whether the key exists and if the type is `[]interface{}`. I'd say if `processors` does not exist then just `return nil` (that's Elasticsearch's problem and it will validate the pipeline). But if it does exist and it's not a `[]interface{}` then that's an error we are forced to handle because we need to iterate over the contents.